src/app/modules/auth/service/auth-guard.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuardService implements CanActivate{
+
+  constructor(private jwtHelper:JwtHelperService) { }
+
+  canActivate(): boolean {
+    const token:any = localStorage.getItem('token');
+
+    if (token==null || this.jwtHelper.isTokenExpired(token)){
+      return false;
+    }
+    
+    return true;
+  }
+}
```

Comment:
In the latest commit i said AuthGuard was implemented, that wasn't AuthGuard that was RoleGuard which job is checking expected role of user. This is real AuthGuard which is checking if user is logged in and if jwt is still valid.